include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -988,13 +988,13 @@ namespace VectorTools
 
 
 
-    template <int dim, typename VectorType, int fe_degree, int n_q_points_1d>
+    template <int dim, typename VectorType, int spacedim, int fe_degree, int n_q_points_1d>
     void project_parallel (const MatrixFree<dim,typename VectorType::value_type> &matrix_free,
                            const ConstraintMatrix &constraints,
                            const std_cxx11::function< VectorizedArray<typename VectorType::value_type> (const unsigned int, const unsigned int)> func,
                            VectorType &vec_result)
     {
-      const DoFHandler<dim> &dof = matrix_free.get_dof_handler();
+      const DoFHandler<dim,spacedim> &dof = matrix_free.get_dof_handler();
```

Comment:
I am confused here:
```
const DoFHandler<dim>& MatrixFree< dim, Number >::get_dof_handler	(	const unsigned int 	fe_component = 0	)	const
```
I though MatrixFree can't work with `dim!=spacedim`. That's why i specifically made functions without `spacedim`.